config/config.go

```diff
@@ -191,6 +196,84 @@ func (cfg Config) CallbackURL(path string, mergeParams ...url.Values) string {
 	return base.String()
 }
 
+//  MatchURL will compare two url strings and will return true if they match.
+func MatchURL(baseURL, testURL string) (bool, error) {
+	compareQueryValues := func(baseVal, testVal url.Values) bool {
+		for name := range baseVal {
+			if baseVal.Get(name) == testVal.Get(name) {
+				continue
+			}
+			return false
+		}
+		return true
+	}
+
+	addImplicitPort := func(u *url.URL) {
+		if strings.Contains(u.Host, ":") {
+			return
+		}
+		switch strings.ToLower(u.Scheme) {
+		case "http":
+			u.Host += ":80"
+		case "https":
+			u.Host += ":443"
+		}
+	}
+	base, err := url.Parse(baseURL)
+	if err != nil {
+		return false, err
+	}
+
+	test, err := url.Parse(testURL)
+	if err != nil {
+		return false, err
+	}
+
+	addImplicitPort(base)
+	addImplicitPort(test)
+
+	// host/port check
+	if !strings.EqualFold(base.Host, test.Host) {
+		return false, nil
+	}
+
+	// scheme check
+	if !strings.EqualFold(base.Scheme, test.Scheme) {
+		return false, nil
+	}
+
+	// path check
+	if len(base.Path) > 1 && !strings.HasPrefix(test.Path, base.Path) {
+		return false, nil
+	}
+
+	// query check
+	if len(test.Query()) > 0 || len(base.Query()) > 0 {
+		if !compareQueryValues(base.Query(), test.Query()) {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+// ValidWebhookURL returns true if the URL is an allowed webhook source.
+func (cfg Config) ValidWebhookURL(whURL string) bool {
+	if len(cfg.Webhook.AllowedURLs) == 0 {
+		return true
+	}
+	for _, u := range cfg.Webhook.AllowedURLs {
+		matched, err := MatchURL(whURL, u)
+		if err != nil {
+			return false
+		}
+		if matched {
+			return matched
```

Comment:
nit: return an explicit `true` here to make the code more scannable (i.e., quick glance to see all possible `return true` locations) -- for the same reason we'd `return value, nil` after an err check instead of `return value, err`
```suggestion
			return true
```